src/org/zaproxy/zap/extension/ascanrulesAlpha/SQLInjectionMsSQL.java

```diff
@@ -0,0 +1,319 @@
+ * 2) Boolean Based (N/A - uses standard syntax)
+ * 3) UNION based (N/A - uses standard syntax)
+ * 4) Stacked (N/A - uses standard syntax)
+ * 5) Blind/Time Based (Yes - uses specific syntax)
+ *
+ * See the following for some great MySQL specific tricks which could be integrated here
+ * http://www.websec.ca/kb/sql_injection#MSSQL_Stacked_Queries
+ * http://pentestmonkey.net/cheat-sheet/sql-injection/mssql-sql-injection-cheat-sheet
+ */
+public class SQLInjectionMsSQL extends AbstractAppParamPlugin {
+
+	private boolean doTimeBased = false;
+	private int doTimeMaxRequests = 0;
+	private int sleep = 5;
+
+	/**
+	 * MSSQL one-line comment
+	 */
+	public static final String SQL_ONE_LINE_COMMENT = " -- ";
+
+	private static final String ORIG_VALUE_TOKEN = "<<<<ORIGINALVALUE>>>>";
+	private static final String SLEEP_TOKEN = "<<<<SLEEP>>>>";
+
+	/**
+	 * create a map of SQL related error message fragments, and map them back to the RDBMS that they are associated with
+	 * keep the ordering the same as the order in which the values are inserted, to allow the more (subjectively judged) common cases to be tested first
+	 * Note: these should represent actual (driver level) error messages for things like syntax error,
+	 * otherwise we are simply guessing that the string should/might occur.
+	 */
+	private static final Map<String, String> SQL_ERROR_TO_DBMS = new LinkedHashMap<>();
+	static {
+		SQL_ERROR_TO_DBMS.put("com.microsoft.sqlserver.jdbc", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("com.microsoft.jdbc", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("com.inet.tds", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("com.microsoft.sqlserver.jdbc", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("com.ashna.jturbo", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("weblogic.jdbc.mssqlserver", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("[Microsoft]", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("[SQLServer]", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("[SQLServer 2000 Driver for JDBC]", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("net.sourceforge.jtds.jdbc", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("80040e14", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("800a0bcd", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("80040e57", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("All queries in an SQL statement containing a UNION operator must have an equal number of expressions in their target lists", "MsSQL");
+		SQL_ERROR_TO_DBMS.put("All queries combined using a UNION, INTERSECT or EXCEPT operator must have an equal number of expressions in their target lists", "MsSQL");
+		//Note: only MSSQL mappings here.
+	}
+
+	/**
+	 * MsSQL specific time based injection strings.
+	 * Note: <<<<ORIGINALVALUE>>>> is replaced with the original parameter value at runtime in these examples below (see * comment)
+	 * TODO: Add SQL_MSSQL_TIME_REPLACEMENTS queries related to PARAM in SELECT/UPDATE/DELETE clause
+	*/
+	private static String[] SQL_MSSQL_TIME_REPLACEMENTS = {
+		ORIG_VALUE_TOKEN + " WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,						//Param in WHERE clause
```

Comment:
Are the comments (at the end of the line) really needed? (Could have just one at the beginning of the constants?)